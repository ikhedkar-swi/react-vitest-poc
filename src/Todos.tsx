import React, { useEffect, useState } from "react";
import "./App.css";
import { getTodos } from "./services/todos.service";

interface Todo {
  userId: number;
  id: number;
  todo: string;
  completed: boolean;
}

export function Todos() {
  const [todos, setTodos] = useState<Todo[]>([]);

  useEffect(() => {
    async function fetchTodos() {
      const todos = await getTodos();
      console.log(todos);
      setTodos(todos.todos);
    }
    fetchTodos();
  }, []);

  return (
    <div>
      <table>
        <tr>
          <th>Id</th>
          <th>UserId</th>
          <th>Todo</th>
          <th>Completed</th>
        </tr>
        {todos.map((todo: Todo) => (
          <tr>
            <td>{todo.id}</td>
            <td>{todo.userId}</td>
            <td>{todo.todo}</td>
            <td>{todo.completed ? "Done" : "In-Progress"}</td>
          </tr>
        ))}
      </table>
    </div>
  );
}

export default Todos;
