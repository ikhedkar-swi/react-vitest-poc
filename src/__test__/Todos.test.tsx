import { expect, test, vi, afterEach } from "vitest";
import renderer from "react-test-renderer";
import Todos from "../Todos";

afterEach(() => {
  vi.restoreAllMocks();
});

global.fetch = vi.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        todos: [
          {
            id: 1,
            todo: "Do something nice for someone ğŸ˜‰",
            completed: true,
            userId: 26,
          },
        ],
        total: 150,
        skip: 0,
        limit: 30,
      }),
  })
);

test("renders todos", async () => {
  let component: any;

  await renderer.act(() => {
    component = renderer.create(<Todos />);
  });

  expect(component.toJSON()).toMatchSnapshot();

  renderer.act(() => {
    component?.unmount();
  });
});
