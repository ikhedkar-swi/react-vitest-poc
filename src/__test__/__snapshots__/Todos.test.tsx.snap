// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders todos 1`] = `
<div>
  <table>
    <tr>
      <th>
        Id
      </th>
      <th>
        UserId
      </th>
      <th>
        Todo
      </th>
      <th>
        Completed
      </th>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        26
      </td>
      <td>
        Do something nice for someone ðŸ˜‰
      </td>
      <td>
        Done
      </td>
    </tr>
  </table>
</div>
`;
